dnf/cli/commands/history.py

```diff
@@ -0,0 +1,506 @@
+                        self._replay_pkg_action(nas[0], action_data)
+                    except TransactionError as e:
+                        errors.append(e)
+            else:
+                try:
+                    self._replay_pkg_action(nas[0], pkg_data)
+                except TransactionError as e:
+                    errors.append(e)
+
+        for group_id, group_data in groups.items():
+            try:
+                action = group_data["action"]
+
+                if action == "Install":
+                    try:
+                        pkg_types = libdnf.transaction.stringToCompsPackageType(group_data["package_types"])
+                    except libdnf.error.Error as e:
+                        errors.append(TransactionError(str(e)))
+                        continue
+
+                    self.base.group_install(group_id, pkg_types)
+                elif action == "Upgrade":
+                    self.base.group_upgrade(group_id)
+                elif action == "Removed":
+                    self.base.group_remove(group_id)
+                else:
+                    errors.append(TransactionError(
+                        _('Unexpected value of group action "{action}" for group "{group}".')
+                            .format(action=action, group=group_id)
+                    ))
+            except KeyError as e:
+                errors.append(TransactionError(
+                    _('Missing object key "{key}" under group "{group}".').format(key=e.args[0], group=group_id)
+                ))
+            except ValueError as e:
+                errors.append(TransactionError(str(e)))
+            except dnf.exceptions.CompsError as e:
+                errors.append(TransactionError(str(e)))
+
+        for env_id, env_data in environments.items():
+            try:
+                action = env_data["action"]
+
+                if action == "Install":
+                    try:
+                        pkg_types = libdnf.transaction.stringToCompsPackageType(env_data["package_types"])
+                    except libdnf.error.Error as e:
+                        errors.append(TransactionError(str(e)))
+                        continue
+
+                    self.base.environment_install(env_id, pkg_types)
+                elif action == "Upgrade":
+                    self.base.environment_upgrade(env_id)
+                elif action == "Removed":
+                    self.base.environment_remove(env_id)
+                else:
+                    errors.append(TransactionError(
+                        _('Unexpected value of environment action "{action}" for environment "{env}".')
+                            .format(action=action, env=env_id)
+                    ))
+            except KeyError as e:
+                errors.append(TransactionError(
+                    _('Missing object key "{key}" under environment "{env}".').format(key=e.args[0], env=env_id)
+                ))
+            except ValueError as e:
```

Comment:
What about to be more flexible with --skip-broken option. This option is very important for any history transaction command because otherwise it will be impossible to use them on Fedora.